### PR TITLE
chore: add prettier with commit hook

### PR DESCRIPTION
**Pull request by jpangelle**
This pull request adds a [prettier](https://prettier.io/) config to the codebase so that the codebase is consistently formatted.

Added three new dependencies:
- `prettier` - the engine to apply formatting
- `husky` - tool to configure `git` hooks
- `pretty-quick` - runs `prettier` on changed files

I configured a pre-commit hook that will automatically apply `prettier` to code changes. The reason this is beneficial is that if a developer has `prettier` turned off in their text editor or maybe doesn't even know what `prettier` is, the formatting will be applied seamlessly in the background when they commit. This ensures that our code is always consistently formatted regardless of a developer's coding environment 😄

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -2,6 +2,27 @@
 # yarn lockfile v1
```
Comment (jpangelle) [inline yarn.lock:2]: This whole file is auto-generated so you can ignore reviewing this.

File: .prettierrc.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+module.exports = {
```
Comment (jpangelle) [inline .prettierrc.js:1]: Below are the rules we are setting for `prettier` to follow.

File: package.json [hunk 3]
```diff
@@ -12,5 +12,15 @@
...
+  },
+  "devDependencies": {
+    "husky": "^4.2.5",
+    "prettier": "1.19.1",
+    "pretty-quick": "^2.0.1"
+  },
+  "husky": {
+    "hooks": {
+      "pre-commit": "pretty-quick --staged"
+    }
```
Comment (jpangelle) [inline package.json:24]: This is our `prettier` pre-commit hook. It applies `prettier` to the code that is changed when you commit. The reason this is beneficial is that if a developer has `prettier` turned off in their text editor or maybe doesn't even know what `prettier` is, the formatting will be applied seamlessly in the background when they commit. Assures that our code is always consistently formatted.

File: package.json [hunk 4]
```diff
@@ -10,7 +10,17 @@
   "author": "Alex Abushanab, JP Angelle",
   "license": "MIT",
   "dependencies": {
-    "nodemon": "2.0.4",
     "robotjs": "0.6.0"
+  },
+  "devDependencies": {
+    "husky": "^4.2.5",
+    "nodemon": "2.0.4",
```
Comment (jpangelle) [inline package.json:17]: Moved `nodemon` to be a `devDependencies` rather than a `dependencies` so it is only used in `development`.

Comment (lexabu) [reply package.json:17]: Good catch.

**Pull request merged**